File: frontend/src/api/query.ts
/**
 * SQL 查询 API
 */

import request from '@/utils/request'
import type { ApiResponse, QueryResult } from '@/types/api'

/**
 * 查询执行请求参数
 */
export interface QueryExecuteRequest {
  connection_id: number
  database?: string
  sql: string
}

/**
 * 查询执行响应（复用 QueryResult 类型）
 */
export type QueryExecuteResponse = QueryResult

/**
 * 查询错误响应
 */
export interface QueryErrorResponse {
  error: string
  query_id?: number
}

/**
 * 执行计划请求参数
 */
export interface ExplainRequest {
  connection_id: number
  database?: string
  sql: string
}

/**
 * 执行计划响应
 */
export interface ExplainResponse {
  plan: Record<string, unknown>[]
  execution_time_ms: number
}

/**
 * 多语句执行请求参数
 */
export interface MultiQueryExecuteRequest {
  connection_id: number
  database?: string
  sql: string
  transaction?: boolean // 是否在事务中执行
}

/**
 * 多语句执行响应
 */
export interface MultiQueryExecuteResponse {
  results: Array<{
    sql: string
    result?: QueryResult
    error?: string
    execution_time_ms: number
  }>
  total_time_ms: number
}

/**
 * 执行 SQL 查询
 */
export async function executeQuery(params: QueryExecuteRequest): Promise<QueryExecuteResponse> {
  const response = await request.post<ApiResponse<QueryExecuteResponse>>('/query/execute', params)
  return response.data
}

/**
 * 获取 SQL 执行计划
 */
export async function explainQuery(params: ExplainRequest): Promise<ExplainResponse> {
  const response = await request.post<ApiResponse<ExplainResponse>>('/query/explain', params)
  return response.data
}

/**
 * 执行多个 SQL 语句
 */
export async function executeMultiQuery(params: MultiQueryExecuteRequest): Promise<MultiQueryExecuteResponse> {
  const response = await request.post<ApiResponse<MultiQueryExecuteResponse>>('/query/execute-multi', params)
  return response.data
}

/**
 * 开始事务
 */
export async function beginTransaction(connectionId: number, database?: string): Promise<void> {
  await request.post<ApiResponse<void>>('/query/transaction/begin', {
    connection_id: connectionId,
    database,
  })
}

/**
 * 提交事务
 */
export async function commitTransaction(connectionId: number, database?: string): Promise<void> {
  await request.post<ApiResponse<void>>('/query/transaction/commit', {
    connection_id: connectionId,
    database,
  })
}

/**
 * 回滚事务
 */
export async function rollbackTransaction(connectionId: number, database?: string): Promise<void> {
  await request.post<ApiResponse<void>>('/query/transaction/rollback', {
    connection_id: connectionId,
    database,
  })
}

File: backend/internal/models/query.go
package models

// QueryExecuteRequest SQL 查询执行请求
type QueryExecuteRequest struct {
	ConnectionID int    `json:"connection_id" binding:"required"`
	Database     string `json:"database"`
	SQL          string `json:"sql" binding:"required"`
	ConfirmDangerous bool `json:"confirm_dangerous,omitempty"`
}

// QueryExecuteResponse SQL 查询执行响应
type QueryExecuteResponse struct {
	Columns            []string        `json:"columns"`
	Rows               [][]interface{} `json:"rows"`
	ExecutionTimeMs    int             `json:"execution_time_ms"`
	RowsAffected       int             `json:"rows_affected"`
	StatementsExecuted int             `json:"statements_executed"`
	QueryID            int             `json:"query_id,omitempty"`
}

// QueryErrorResponse SQL 查询错误响应
type QueryErrorResponse struct {
	Error   string `json:"error"`
	QueryID int    `json:"query_id,omitempty"`
}

// ExplainRequest SQL 执行计划请求
type ExplainRequest struct {
	ConnectionID int    `json:"connection_id" binding:"required"`
	Database     string `json:"database"`
	SQL          string `json:"sql" binding:"required"`
}

// ExplainResponse SQL 执行计划响应
type ExplainResponse struct {
	Plan            []map[string]interface{} `json:"plan"`
	ExecutionTimeMs int                      `json:"execution_time_ms"`
}

// TransactionRequest 事务操作请求
type TransactionRequest struct {
	ConnectionID int    `json:"connection_id" binding:"required"`
	Database     string `json:"database"`
}
